crates/bpf-builder/include/output.bpf.h

```diff
@@ -16,15 +23,65 @@
     struct buffer buffer;                                                      \
   };                                                                           \
                                                                                \
-  static __always_inline struct struct_name *struct_name##_init(               \
-      int event_variant, struct bpf_map_def_aya *tracker) {                    \
-    pid_t tgid = tracker_interesting_tgid(tracker);                            \
-    if (tgid < 0)                                                              \
-      return NULL;                                                             \
+  /* The BPF stack limit of 512 bytes is exceeded by network_event,*/          \
+  /* so we use a per-cpu array as a workaround*/                               \
+  struct {                                                                     \
+    __uint(type, BPF_MAP_TYPE_PERCPU_ARRAY);                                   \
+    __type(key, u32);                                                          \
+    __type(value, struct struct_name);                                         \
+    __uint(max_entries, MAX_PREEMPTION_NESTING_LEVEL);                         \
+  } map_temp_##struct_name SEC(".maps");                                       \
+                                                                               \
+  struct {                                                                     \
+    __uint(type, BPF_MAP_TYPE_PERCPU_ARRAY);                                   \
+    __type(key, u32);                                                          \
+    __type(value, u64);                                                        \
+    __uint(max_entries, 1);                                                    \
+  } map_nesting_##struct_name SEC(".maps");                                    \
+                                                                               \
+  static __always_inline void decrease_nesting_##struct_name() {               \
+    u32 zero = 0;                                                              \
+    u64 *nesting_level =                                                       \
+        bpf_map_lookup_elem(&map_nesting_##struct_name, &zero);                \
+    if (nesting_level) {                                                       \
+      sync_decrement(nesting_level);                                           \
+    }                                                                          \
+  }                                                                            \
```

Comment:
I didn't add it because it implies the API has been used wrong (eg. it could be a Rust panic).
I'm adding it for consistency (and to make it easier to spot such an API usage error)